### PR TITLE
JsDocstrings in spidermonkey and functions for os.readfile(), os.writefile(), os.getHostByName(), os.system(), eval()

### PR DESCRIPTION
**Pull request by verpeteren**
1) I have added the documentation to the source so that it is easier to maintain and run and update the website-documentation.
2) I grouped moved the getHostByName and the readfile functions out the Ape namespace and put them into the Ape.os namespace.
3) I added a os.system, os.writefile, eval function
4) fixed some typo's, cleared some whitespace, added some todo, fixed another problem in the build.sh
5) fixed a bug with a missing fallthrough

### CODE REVIEW

**Comment by lcharette** [comment]
Is the "os" thing for new function or old one? If this is changed, could it break compatibility with old ssjs?

File: modules/libape-spidermonkey.c [hunk 1]
```diff
@@ -3186,7 +5064,6 @@ static struct _ape_mysql_queue *apemysql_push_queue(struct _ape_mysql_data *myha
 	JS_FS("getUserByPubid", ape_sm_get_user_by_pubid, 1, 0),
 	JS_FS("getChannelByPubid", ape_sm_get_channel_by_pubid, 1, 0),
 	JS_FS("config", ape_sm_config, 2, 0),
-	JS_FS("getHostByName", ape_sm_gethostbyname, 1, 0),
```
Comment (verpeteren) [inline modules/libape-spidermonkey.c:3189]: keep this line, to keep the api stable

File: modules/libape-spidermonkey.c [hunk 2]
```diff
@@ -3196,9 +5073,17 @@ static struct _ape_mysql_queue *apemysql_push_queue(struct _ape_mysql_data *myha
 	JS_FS("addUser", ape_sm_adduser, 1, 0),
 	JS_FS("mkChan", ape_sm_mkchan, 1, 0),
 	JS_FS("rmChan", ape_sm_rmchan, 1, 0),
-	JS_FS("system", ape_sm_system, 2, 0),
 	JS_FS("eval", ape_sm_eval, 1, 0),
-	JS_FS("readfile", ape_sm_readfile, 1, 0),
```
Comment (verpeteren) [inline modules/libape-spidermonkey.c:3201]: keep this line to keep the api stable

File: modules/libape-spidermonkey.c [hunk 3]
```diff
@@ -1919,52 +2959,146 @@ static unsigned int ape_sm_cmd_wrapper(callbackp *callbacki)
...
+ * @public
+ * @static
+ *
+ * @param {string} filename The filename to read
+ * @returns {string} The content of the file or NULL
+ *
+ * @example
+ * var content = os.readfile('/etc/hosts');
+ */
 APE_JS_NATIVE(ape_sm_readfile)
```
Comment (verpeteren) [inline modules/libape-spidermonkey.c:3075]: /**
- Get the content of a file.
  *
- @name Ape.readfile
- @function
- @public
- @static
- @deprecated Please use Ape.os.readfile
  *
- @param {string} filename The filename to read
- @returns {string} The content of the file or NULL
  *
- @example
- var content = Ape.readfile('/etc/hosts');
  *
- @see Ape.os.readfile
  */

File: modules/libape-spidermonkey.c [hunk 4]
```diff
@@ -2382,15 +3852,16 @@ static JSObject *get_pipe_object(const char *pubid, transpipe *pipe, JSContext *
...
+ * @param {string} hostname The Hostname
+ * @returns {string} ip		The ip address af the hostname or NULL
+ *
  * @example
- * var content = Ape.getHostByName("www.ape-project");
- * @name Ape.getHostByName
- * @function
+ * var content = os.getHostByName("www.ape-project");
  */
 APE_JS_NATIVE(ape_sm_gethostbyname)
```
Comment (verpeteren) [inline modules/libape-spidermonkey.c:3987]: /\* @name Ape.getHostByName
- @function
- @public
- @static
- @deprecated Please use Ape.os.getHostByName
  *
- @param {string} hostname The Hostname
- @returns {string} ip     The ip address af the hostname or NULL
  *
- @example
- var content = Ape.getHostByName('www.ape-project');
  *
- @see Ape.os.getHostByName
  */

**Comment by verpeteren** [comment]
Well, in theory it could be a api breaker, although I am not sure if anyone used them because:
- There was a undocumented Ape.readfile function that had a lot of printfs in it.
- The Ape.getHostByName was just introduced by me, and was in a previous pull request.
  It is easy to keep the api going by adding the following 2 lines to the "static JSFunctionSpec ape_funcs[] = {" block again.

JS_FS("getHostByName", ape_sm_gethostbyname, 1, 0),
JS_FS("readfile", ape_sm_readfile, 1, 0),

And some documentation as well

/\* @name Ape.getHostByName
- @function
- @public
- @static
- @deprecated Please use Ape.os.getHostByName
  *
- @param {string} hostname The Hostname
- @returns {string} ip     The ip address af the hostname or NULL
  *
- @example
- var content = Ape.getHostByName('www.ape-project');
  *
- @see Ape.os.getHostByName
  */

/**
- Get the content of a file.
  *
- @name Ape.readfile
- @function
- @public
- @static
- @deprecated Please use Ape.os.readfile
  *
- @param {string} filename The filename to read
- @returns {string} The content of the file or NULL
  *
- @example
- var content = Ape.readfile('/etc/hosts');
  *
- @see Ape.os.readfile
  */

Should I create a new commit/pull request? How does this work?
I added some notes to the diff here on github.

**Comment by paraboul** [comment]
using getHostByName is very dangerous. The POSIX API gethostbyname is blocking. That's why APE internally uses udns in order to resolve host. That is, if a resolution takes 5sec, the entire server is going to hang for 5 sec.

**Comment by verpeteren** [comment]
I wanted to use getHostName as a poor-mans High-Availibitiy solution, because Ape.MySQL needs an IP, and not a hostname. And if the errorcode after a query indicates a broken connection, I did a getHostByName and reconnect again 

But I understand the concerns and I played around alittle bit; and I am almost there.
I will write a udns variant: something that can work like: 

Ape.os.resolveHostByName('www.ape-project.com', function (ip) {
  Ape.log('resolved: ' + ip);
  var sql = new Ape.MySQL( ip + ':3306', 'user', 'password', 'database');
         sql.onConnect = function() {
               Ape.log('Connected to mysql server: ' + ip);
         }
   });
});

I will come back on this in a few days.

**Comment by verpeteren** [comment]
I have closed this pull request. I will reorganise my changes and make another one.

**Pull request closed without merging**